source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2906,6 +2700,7 @@ bool COpenGLDriver::setRenderTarget(IFrameBuffer* frameBuffer, bool setNewViewpo
     if (found->CurrentFBO)
         found->CurrentFBO->drop();
     found->CurrentFBO = static_cast<COpenGLFrameBuffer*>(frameBuffer);
+    //TODO what was it supposed to be used for? I need to know if i have to do somehing with GL state here
```

Comment:
its just a flush afaik

Reply:
flush of a current state set to the default state

was actually to implicitly enable depth and color masks as well as disable rasterizer discard ( a few things that would mess with framebuffer clearing and blitting)

Reply:
the optimize comment is about the fact that not everything really needed to be reset to default

Reply:
it comes down to this https://github.com/buildaworldnet/IrrlichtBAW/blob/master/source/Irrlicht/COpenGLSLMaterialRenderer.cpp#L229

Reply:
so its just setting a few blending parameters. Is it even worth doing anything? especially hen we dont have `E_MATERIAL_TYPE` (`EMT_*`) any more ? + you sure it means "reset to default" and not "run some GL calls again no matter what"?

Reply:
it means reset to default. its to avoid a nasty suprise when clearing the Color and Depth/Stencil attachments.

You can read glClearBuffer* and `glBlit*` documentation and see that `glEnable(GL_RASTERIZER_DISCARD)` as well as depth,stenil,and color masks affect which channels get cleared if any (`glClearBuffer*` actually gets treated like a render call).

The parameter here used to be "reset everything to default" but it as obviously overkill.

Alternatively instead of resetting this stuff here, you could reset it in the Clear and Blit functions (then reset back to the original in the next draw).

Reply:
@Crisspl maybe best to reset in Clear and Blit functions  (then reset back to the original in the next draw).

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r328375630

Reply:
resolved

Reply:
TODO `clearScreen()` see comments